server/compose/service/record.go

```diff
@@ -714,6 +719,133 @@ func (svc record) Bulk(ctx context.Context, skipFailed bool, oo ...*types.Record
 	}
 }
 
+func (svc record) UpdateAll(ctx context.Context, namespaceID uint64, moduleID uint64, values types.RecordValueSet, recordIds ...uint64) (err error) {
+	var (
+		f = types.RecordFilter{
+			NamespaceID: namespaceID,
+			ModuleID:    moduleID,
+			Deleted:     filter.State(0),
+		}
+	)
+
+	return svc.processUpdateAll(ctx, f, values, types.OperationTypePatch, recordIds)
+}
+
+func (svc record) DeleteAll(ctx context.Context, namespaceID uint64, moduleID uint64, recordIds ...uint64) (err error) {
+	var (
+		f = types.RecordFilter{
+			NamespaceID: namespaceID,
+			ModuleID:    moduleID,
+			Deleted:     filter.State(0),
+		}
+	)
+
+	return svc.processUpdateAll(ctx, f, nil, types.OperationTypeDelete, recordIds)
+}
+
+func (svc record) UndeleteAll(ctx context.Context, namespaceID uint64, moduleID uint64, recordIds ...uint64) (err error) {
+	var (
+		f = types.RecordFilter{
+			NamespaceID: namespaceID,
+			ModuleID:    moduleID,
+			Deleted:     filter.State(1),
+		}
+	)
+
+	return svc.processUpdateAll(ctx, f, nil, types.OperationTypeUndelete, recordIds)
+}
+
+func (svc record) processUpdateAll(ctx context.Context, f types.RecordFilter, values types.RecordValueSet, operation types.OperationType, excludeIds []uint64) (err error) {
+	var (
+		ns           *types.Namespace
+		m            *types.Module
+		r            *types.Record
+		records      types.RecordSet
+		recordFilter types.RecordFilter
+
+		aProps = &recordActionProps{
+			namespace: &types.Namespace{ID: f.NamespaceID},
+			module:    &types.Module{ID: f.ModuleID},
+		}
+		action func(props ...*recordActionProps) *recordAction
+
+		valueError *types.RecordValueErrorSet
+	)
+
+	return store.Tx(ctx, svc.store, func(ctx context.Context, s store.Storer) error {
+		// load both the namespace and module
+		if ns, m, err = loadModuleCombo(ctx, svc.store, f.NamespaceID, f.ModuleID); err != nil {
+			return err
+		}
+
+		aProps.setNamespace(ns)
+		aProps.setModule(m)
+
+		f.Limit = 10
```

Comment:
Why did you decide to go with such a low number?
I usually go with 300-500 (that was the sweet spot for me).